nav2_theta_star_planner/src/theta_star_planner.cpp

```diff
@@ -0,0 +1,142 @@
+{
+  planner_ = std::make_unique<theta_star::ThetaStar>();
+  auto node = parent.lock();
+  logger_ = node->get_logger();
+  clock_ = node->get_clock();
+  name_ = name;
+  tf_ = tf;
+  planner_->costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".how_many_corners", rclcpp::ParameterValue(8));
+
+  node->get_parameter(name_ + ".how_many_corners", planner_->how_many_corners_);
+
+  if (planner_->how_many_corners_ != 8 && planner_->how_many_corners_ != 4) {
+    planner_->how_many_corners_ = 8;
+    RCLCPP_WARN(logger_, "Your value for - .how_many_corners  was overridden, and is now set to 8");
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_euc_cost", rclcpp::ParameterValue(4.0));
+  node->get_parameter(name_ + ".w_euc_cost", planner_->w_euc_cost_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_traversal_cost", rclcpp::ParameterValue(7.0));
+  node->get_parameter(name_ + ".w_traversal_cost", planner_->w_traversal_cost_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_heuristic_cost", rclcpp::ParameterValue(1.0));
+  node->get_parameter(name_ + ".w_heuristic_cost", planner_->w_heuristic_cost_);
+}
+
+void ThetaStarPlanner::cleanup()
+{
+  RCLCPP_INFO(logger_, "CleaningUp plugin %s of type nav2_theta_star_planner", name_.c_str());
+}
+
+void ThetaStarPlanner::activate()
+{
+  RCLCPP_INFO(logger_, "Activating plugin %s of type nav2_theta_star_planner", name_.c_str());
+}
+
+void ThetaStarPlanner::deactivate()
+{
+  RCLCPP_INFO(logger_, "Deactivating plugin %s of type nav2_theta_star_planner", name_.c_str());
+}
+
+nav_msgs::msg::Path ThetaStarPlanner::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  nav_msgs::msg::Path global_path;
+  auto start_time = std::chrono::steady_clock::now();
+  planner_->setStartAndGoal(start, goal);
+  RCLCPP_DEBUG(
+    logger_, "Got the src and dst... (%i, %i) && (%i, %i)",
+    planner_->src_.x, planner_->src_.y, planner_->dst_.x, planner_->dst_.y);
+  getPlan(global_path);
+  auto stop_time = std::chrono::steady_clock::now();
+  auto dur = std::chrono::duration_cast<std::chrono::microseconds>(stop_time - start_time);
+  RCLCPP_DEBUG(logger_, "the time taken is : %i", static_cast<int>(dur.count()));
+  return global_path;
+}
+
+void ThetaStarPlanner::getPlan(nav_msgs::msg::Path & global_path)
+{
+  std::vector<coordsW> path;
+
+  if (planner_->isUnsafeToPlan()) {
+    RCLCPP_ERROR(logger_, "Either of the start or goal pose are an obstacle! ");
+    global_path.poses.clear();
+  } else if (planner_->generatePath(path)) {
+    global_path = linearInterpolation(path, planner_->costmap_->getResolution());
+  } else {
+    RCLCPP_ERROR(logger_, "Could not generate path between the given poses");
+    global_path.poses.clear();
+  }
+  path.clear();
```

Comment:
path.clear() isnt necessary, it will be destructed at the end of this scope and isn't used anywhere else after this